copylot/assemblies/photom/demo/demo_roi.py

```diff
@@ -0,0 +1,314 @@
+import sys
+import logging
+from PyQt5.QtWidgets import QApplication, QMainWindow, QComboBox, QVBoxLayout, QWidget, QPushButton, QLabel, QLineEdit, QHBoxLayout, QSpacerItem, QSizePolicy
+from PyQt5.QtCore import Qt, QPoint, QRect, pyqtSignal
+from PyQt5.QtGui import QPainter, QPen, QPainterPath, QPolygon, QMouseEvent, QPaintEvent
+from copylot.assemblies.photom.utils.pattern_tracing import Shape
+
+logging.basicConfig(level=logging.DEBUG, format='%(asctime)s - %(levelname)s - %(message)s')
+
+
+class ViewerWindow(QMainWindow):
+    shapesUpdated = pyqtSignal()
+
+    def __init__(self) -> None:
+        """initializes the Viewer Window for drawing and viewing shapes.
+        """
+        super().__init__()
+        self.setWindowTitle("Main Window")
+        self.setGeometry(450, 100, 400, 300)
+
+        # class variables
+        self.drawing = False
+        self.shapes = {}
+        self.curr_shape_points = []
+        self.curr_shape_id = 0
+        self.selected_shape_id = None
+
+    def mousePressEvent(self, event: QMouseEvent) -> None:
+        """records left mouse press for drawing shapes.
+
+        Args:
+            event (QMouseEvent): the mouse event that triggered the press. 
+        """
+        if event.button() == Qt.LeftButton:
+            point = event.pos()
+            if not self.drawing:
+                self.drawing = True
+            self.curr_shape_points.append(point)
+
+    def mouseMoveEvent(self, event: QMouseEvent) -> None:
+        """records mouse movement while drawing a shape.
+
+        Args:
+            event (QMouseEvent): the mouse event that triggered the movement. 
+        """
+        if self.drawing:
+            point = event.pos()
+            self.curr_shape_points.append(point)
+            self.update()
+
+    def mouseReleaseEvent(self, event: QMouseEvent) -> None:
+        """handles the release of the left mouse button, completing the drawing of a shape.
+
+        Args:
+            event (QMouseEvent): the mouse event that triggered the release. 
+        """
+        if event.button() == Qt.LeftButton:
+            if self.drawing:
+                point = event.pos()
+                self.curr_shape_points.append(point)
+
+                self.curr_shape_points.append(self.curr_shape_points[0]) # connecting the final points in case not connected
```

Comment:
it might be that this needs to be abstracted a bit so it's easier to read. You will need to add this to the `eventFilter`in windows.py